packages/decentraland-ecs/src/decentraland/UIShapes.ts

```diff
@@ -350,17 +350,36 @@ export class UIInputText extends UIShape {
   @ObservableComponent.field
   paddingLeft: number = 0
 
-  @OnUUIDEvent.uuidEvent
   onTextSubmit: OnTextSubmit | null = null
 
-  @OnUUIDEvent.uuidEvent
   onChanged: OnChanged | null = null
 
+  // @internal
+  @OnUUIDEvent.uuidEvent
+  protected readonly onTextChanged: OnChanged
+
   @OnUUIDEvent.uuidEvent
   onFocus: OnFocus | null = null
 
   @OnUUIDEvent.uuidEvent
   onBlur: OnBlur | null = null
+
+  constructor(parent: UIShape | null) {
+    super(parent)
+    this.onTextChanged = new OnChanged((e) => {
+      const { value, isSubmit } = e.value
+
+      const isDirty = this.dirty
+      this.value = value
+      this.dirty = isDirty
```

Comment:
i don't understand this dirty thing.
Why assign to a new prop and then reassign it to this.dirty ?

Reply:
modifying `this.value` will set component as `dirty`.
`dirty` component will send a message with it values to the renderer.
since `this.value` here is being set by a message from the renderer itself, we don't want to send a message back, unless something else than `this.value` have changed.